url0d38a10ee7_diegopliebana_GEBT_tgz-pJ8-ch_idsia_scenarios_test_EvolveWithChangingSeedsJ8/src/ch/idsia/mario/engine/BgRenderer.java

```diff
@@ -0,0 +1,101 @@
+package ch.idsia.mario.engine;
+
+import java.awt.*;
+import java.util.Random;
+import ch.idsia.mario.engine.level.*;
+
+
+public class BgRenderer
+{
+    private int xCam;
+    private int yCam;
+    private Image image;
+    private Graphics2D g;
+    private static final Color transparent = new Color(0, 0, 0, 0);
+    private Level level;
+
+    private Random random = new Random();
+    public boolean renderBehaviors = false;
+
+    private int width;
+    private int height;
+    private int distance;
+
+    public BgRenderer(Level level, GraphicsConfiguration graphicsConfiguration, int width, int height, int distance)
+    {
+        this.distance = distance;
+        this.width = width;
+        this.height = height;
+
+        this.level = level;
+        image = graphicsConfiguration.createCompatibleImage(width, height, Transparency.BITMASK);
+        g = (Graphics2D) image.getGraphics();
+        g.setComposite(AlphaComposite.Src);
+
+        updateArea(0, 0, width, height);
+    }
+
+    public void setCam(int xCam, int yCam)
+    {
+        xCam /= distance;
+        yCam /= distance;
+        int xCamD = this.xCam - xCam;
+        int yCamD = this.yCam - yCam;
+        this.xCam = xCam;
+        this.yCam = yCam;
+
+        g.setComposite(AlphaComposite.Src);
+        g.copyArea(0, 0, width, height, xCamD, yCamD);
+
+        if (xCamD < 0)
+        {
+            if (xCamD < -width) xCamD = -width;
+            updateArea(width + xCamD, 0, -xCamD, height);
+        }
+        else if (xCamD > 0)
+        {
+            if (xCamD > width) xCamD = width;
+            updateArea(0, 0, xCamD, height);
+        }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 50:59, 61:70.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.